contracts/DataAvailability.sol

```diff
@@ -0,0 +1,223 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.24;
+
+import {AccessControlDefaultAdminRules} from '@openzeppelin/contracts/access/extensions/AccessControlDefaultAdminRules.sol';
+
+contract DataAvailability is AccessControlDefaultAdminRules {
+  /**
+     * @dev DataAvailability stores information about the objective function value for users
```

Comment:
Can you give context/example on how a reward producer might use this?  E.g., "deploy a new instance for each 'KPI'"